src/fileManager.ts

```diff
@@ -1,114 +1,660 @@
-import { BatchId } from '@ethersphere/bee-js';
+import {
+  BatchId,
+  Bee,
+  BeeRequestOptions,
+  Bytes,
+  DownloadOptions,
+  EthAddress,
+  GetGranteesResult,
+  GranteesResult,
+  MantarayNode,
+  NodeAddresses,
+  NULL_TOPIC,
+  PostageBatch,
+  PrivateKey,
+  PssSubscription,
+  RedundancyLevel,
+  Reference,
+  STAMPS_DEPTH_MAX,
+  Topic,
+  UploadOptions,
+  Utils,
+} from '@upcoming/bee-js';
+import path from 'path';
 
-import { FILE_INFO_LOCAL_STORAGE } from './constants';
-import { FileInfo, ShareItem } from './types';
+import {
+  FILE_INFO_LOCAL_STORAGE,
+  OWNER_FEED_STAMP_LABEL,
+  REFERENCE_LIST_TOPIC,
+  SHARED_INBOX_TOPIC,
+  SWARM_ZERO_ADDRESS,
+} from './utils/constants';
+import { BeeVersionError, FileInfoError, SignerError, StampError } from './utils/errors';
+import { FetchFeedUpdateResponse, FileInfo, ReferenceWithHistory, ShareItem, WrappedFileInfoFeed } from './utils/types';
+import {
+  assertFileInfo,
+  assertShareItem,
+  assertWrappedFileInoFeed,
+  getRandomTopic,
+  isNotFoundError,
+  makeBeeRequestOptions,
+  makeNumericIndex,
+  numberToFeedIndex,
+  readFile,
+} from './utils/utils';
 
 export class FileManager {
+  private bee: Bee;
+  private signer: PrivateKey;
+  private nodeAddresses: NodeAddresses;
+  private stampList: PostageBatch[];
+  private ownerFeedList: WrappedFileInfoFeed[];
   private fileInfoList: FileInfo[];
+  private ownerFeedNextIndex: number;
+  private sharedWithMe: ShareItem[];
+  private sharedSubscription: PssSubscription | undefined;
+  private ownerFeedTopic: Topic;
 
-  constructor() {
+  constructor(bee: Bee) {
+    this.bee = bee;
+    if (!this.bee.signer) {
+      throw new SignerError('Signer required');
+    }
+    this.signer = this.bee.signer;
+    this.stampList = [];
     this.fileInfoList = [];
-
-    // We said that constructor will load the file info list into state, but this only works as long as initialize is synchronous.
+    this.ownerFeedList = [];
+    this.ownerFeedNextIndex = 0;
+    this.ownerFeedTopic = NULL_TOPIC;
+    this.sharedWithMe = [];
+    this.sharedSubscription = undefined;
   }
 
+  // Start init methods
   async initialize(): Promise<void> {
+    await this.verifySupportedVersions();
+    await this.initNodeAddresses();
+    await this.initStamps();
+    await this.initOwnerFeedTopic();
+    await this.initOwnerFeedList();
     await this.initFileInfoList();
   }
 
-  private async initFileInfoList(): Promise<void> {
-    const rawData = localStorage.getItem(FILE_INFO_LOCAL_STORAGE);
-    if (!rawData) {
-      console.error('No data found in data.txt (localStorage');
+  // TODO: is exact version check necessary?
```

Comment:
Check this comment: https://github.com/ethersphere/bee-js/blob/8c801e9d07280939f7448df5f5798883a401af76/src/modules/debug/status.ts#L90

This is way too strict with ACT in mind we need at least 7.0+ bee API.
Right now I cant run the tests with dev node from the master branch.